### PR TITLE
Minimal example of a software supply chain

### PR DESCRIPTION
**Pull request by trishankatdatadog**

### CODE REVIEW

**Comment by radu-matei** [comment]
This is excellent!
I would assume we want to also add a document / update the readme before merging, right?

File: scripts/in-toto/keys.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+# 3rd-party.
+from securesystemslib.interface import (
+    generate_and_write_ed25519_keypair,
+    import_ed25519_privatekey_from_file,
+    import_ed25519_publickey_from_file,
+)
+
+# Constants.
+# Where we use by default to store Notary private keys.
+KEYSTORE_DIR = os.path.expanduser('~/.signy/private')
```
Comment (radu-matei) [inline scripts/in-toto/keys.py]: IIRC, Signy has a configurable config directory - do we want to keep the same convention?

**Comment by radu-matei** [comment]
This is excellent!
I would assume we want to also add a document / update the readme before merging, right?

**Comment by radu-matei** [comment]
This is of course non-blocking for this PR - but is there anything used here (potentially in the `securesystemslib` wheel) that is not implemented in Go?

File: scripts/in-toto/minimal.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        # Machines MUST produce the following products.
+        products = [
+            # 1. Machines MUST produce the same product developers produced.
+            MATCH_MATERIALS(
+                pattern = BUNDLE_ALL,
+                step_name = this_step_name,
+            ),
+            # 2. However, machines MUST fill in ONLY image digests and sizes.
+            # TODO: double-check pattern.
+            MODIFY(f'{BUNDLE_ROOT}.images.*'),
```
Comment (adityasaky) [inline scripts/in-toto/minimal.py:58]: Note: needs `DISALLOW *` here and elsewhere.

**Comment by trishankatdatadog** [comment]
Ok, I consider this case closed. Can I get a ✅?

> I would assume we want to also add a document / update the readme before merging, right?

Not yet: there will be another PR that will make use of these scripts.

**Review by radu-matei — APPROVED** [review]

**Pull request merged**